webpack.config.js

```diff
@@ -0,0 +1,115 @@
+const CopyPlugin = require('copy-webpack-plugin');
+const exec = require('child_process').exec;
+const config = require('./gulp/config');
+
+const outputPath = path.resolve(`${config.outputPath}/${config.jsDirectory}`);
+const jsSource = `./${config.sourcePath}/${config.jsDirectory}`;
+const cleanPaths = [
+  `${config.outputPath}/fonts/*`,
+  `${config.outputPath}/images/*`,
+  `${config.outputPath}/stylesheets/*`,
+  `${config.outputPath}/javascripts/*`
+];
+const entries = {
+  main: [`${jsSource}/main.js`],
+  raven: [`${jsSource}/raven.js`],
+  styleguide: [`${jsSource}/styleguide/styleguide.jsx`],
+  survey: [`${jsSource}/surveys/survey.js`],
+  survey_admin: [`${jsSource}/surveys/survey-admin.js`],
+  survey_results: [`${jsSource}/surveys/survey-results.jsx`],
+  campaigns: [`${jsSource}/campaigns.js`],
+  charts: [`${jsSource}/charts.js`],
+  tinymce: [`${jsSource}/tinymce.js`],
+  embed_course_stats: [`${jsSource}/embed_course_stats.js`]
+};
+
+const wpPlugins = [];
+const absoluteOutputPath = path.join(process.cwd(), config.outputPath);
+const jqueryUlsPath = './node_modules/jquery/dist/jquery.min.js';
+const mode = 'production';
+
+wpPlugins.push(new CleanWebpackPlugin({
+  cleanOnceBeforeBuildPatterns: cleanPaths.map(relative => path.join(process.cwd(), relative))
+}));
+
+wpPlugins.push({
+  apply: (compiler) => {
+    compiler.hooks.afterEmit.tap('AfterEmitPlugin', (_) => {
+      exec('bundle exec rails i18n:js:export', (err, stdout, stderr) => {
+        if (stdout) process.stdout.write(stdout);
+        if (stderr) process.stderr.write(stderr);
+      });
+      require('child_process').spawn('gulp', ['build'], { stdio: 'inherit' });
+    });
+  }
+});
+
+wpPlugins.push(new CopyPlugin({
+  patterns: [{
+    from: jqueryUlsPath, to: `${absoluteOutputPath}/${config.jsDirectory}`
+  },
+  {
+    from: `${config.sourcePath}/${config.imagesDirectory}`, to: `${absoluteOutputPath}/${config.imagesDirectory}`
+  }, {
+    from: `${config.sourcePath}/${config.fontsDirectory}`, to: `${absoluteOutputPath}/${config.fontsDirectory}`
+  }, {
+    from: './node_modules/tinymce/skins', to: `${absoluteOutputPath}/${config.jsDirectory}/skins`
+  }]
+}));
+
+wpPlugins.push(new ManifestPlugin({
+  fileName: 'rev-manifest.json'
+}));
+
+
+wpPlugins.push(new webpack.DefinePlugin({
+  'process.env': {
+    NODE_ENV: JSON.stringify(mode)
+  }
+}));
+
+module.exports = {
+  mode,
+  entry: entries,
+  output: {
+    path: outputPath,
+    filename: '[name].[chunkhash].js',
+    publicPath: '/'
+  },
+  resolve: {
+    extensions: ['.js', '.jsx'],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
+        exclude: [/vendor/, /node_modules(?!\/striptags)/],
+        use: {
+          loader: 'babel-loader',
+          query: {
+            cacheDirectory: true
+          }
+        }
+      },
+      {
+        test: /\.jsx?$/,
+        exclude: [/vendor/, /node_modules(?!\/striptags)/],
+        loader: 'eslint-loader',
```

Comment:
What does this do? I assume the version bump of eslint was required for this?

Reply:
gulp used to do the linting. Now Webpack does.